storage/entities/base_entity.js

```diff
@@ -226,6 +236,74 @@ class BaseEntity {
 		);
 	}
 
+	/**
+	 * Validate allowed filters
+	 * @param {Array.<Object>|Object} filters
+	 * @param {Boolean} required
+	 * @return {true, NonSupportedFilterTypeError>}
+	 */
+	validateFilters(filters = {}, required = false) {
```

Comment:
isn't checking `required`, responsibility of caller?
When you cal function, u are passing filter and saying this is required, but caller is supplying so it sounds redundant.
also, checking `!filters` doesn't make sense because it has default value.
Same for validateOptions.

Reply:
I've added this `required` flag because in `isPersisted` we want to make sure you're passing at least one filter otherwise the method doesn't make sense. Maybe the name of the param is not clear. I'll update it.

The `!filters` was for the case where `filters` is empty string but I've just realised `Object.keys(filters)` will cater for this scenario as well. I'll remove the `!filter` check ;)

Reply:
Actually the `!filters` check if required.
[This test](https://github.com/LiskHQ/lisk/blob/2615-create-block-entity/test/unit/storage/entities/block.js#L550) doesn't pass if I remove it.

```
AssertionError: expected [Function] to throw 'NonSupportedFilterTypeError' but 'TypeError: Cannot convert undefined or null to object' was thrown
```